Automattic/pocket-casts-ios
# Apple SSO: Corrupt Token Failure

| 📘 Project: #381 | Depends on: https://github.com/Automattic/pocket-casts-ios/pull/388 |
|:---:|:---:|

This handles the path for when a user's Auth token has been revoked, and the app cannot log them in again. Previously the app navigated the user to the Password login flow. Now, it will take the user to the start of the login flow so they can select their social login provider again. 

## To test

📓  The simulator doesn't always return the email address so please test with a physical device to ensure every thing works as expected 

1. Enable the Feature Flag `signInWithApple`
2. Switch your Scheme to `Staging`
3. Login with SSO 
4. Navigate to the new developer menu
    - Profile > Settings > Developer
    - 🗒️ This Menu will only appear in Debug and Staging
5. Select "Corrupt Sync Login Token" 
6. Trigger an Authenticated request like Refresh
7. Expect the Token to be restored.
8. Corrupt your token again
9. Remove Apple Sign-In from the app.
    - Apple Settings App > Your Name > Password & Security > Apps Using Apple ID > Pocket Casts > Stop Using Apple ID
10. Return to the App 
11. Expect to see the logged out message with the prompt to sign in
<img width="300" src="https://user-images.githubusercontent.com/3384451/195666396-54fa72f1-c856-46d6-bcc8-8f044b3c065e.png">

12. Tapping Sign-In should take you to step 1 of the authentication flow

## Checklist
